cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/ProgramLifecycleHttpHandlerInternal.java

```diff
@@ -72,8 +72,12 @@ public void getProgramRunRecordMeta(HttpRequest request, HttpResponder responder
                                       @PathParam("program-type") String type,
                                       @PathParam("program-name") String programName,
                                       @PathParam("run-id") String runid) throws Exception {
+    ApplicationId applicationId = new ApplicationId(namespaceId, appName, appVersion);
+    if (ApplicationId.DEFAULT_VERSION.equals(appVersion)) {
+      applicationId = programLifecycleService.getLatestApplicationId(applicationId.getAppReference());
```

Comment:
The latest version will not always match the app version for the given run-id. Is the desired behavior here to return a 404 in that case? Or do we want to always return the run id and ignore the app version?

Reply:
I think we should return the runRecord ignoring the app version. But also we need to return 404 when the app is not found